### ISSUE TITLE
[Feature]: Add CLI option to leave headed browser open after running test

### ISSUE THREAD

**Issue by f-w** [microsoft/playwright#31329]
### 🚀 Feature Request

Add an option for example ``--leave-browser-open``. When specified, browser will be left open after running test(s).  The option is only effective in conjunction with `--headed` option. 

### Example

on OOTB Playwright project, runing 
```
npx playwright test --headed --leave-browser-open --project=chromium example
```
leaves chromium browser open.

### Motivation

A script may have 2 use cases 
1. fully automated regression test
2. as a booster to manual UAT

For example, a CRM web app typically requires login, creating a contact, then a service request. A script automating such workflow can be used alone by itself, or it can be used by a manual UAT tester to save time on these repetitive activities at the begining, then contintue with subsequent manual workflow. In the latter case, browser session needs to be preserved after testing is complete.

Adding manual pause to script won't work because case 1 is also impacted.

Browser will be left open by default when launched by VS Code. But VS Code should be optional, especially for manual UAT tester.

**Comment by dgozman** [comment]
@f-w It seems like you already know about the available alternatives - VSCode extension, `page.pause()`, etc. I do not really understand why it does not work for you.

Just in case, the following snippet that uses environment variables should be 100% equivalent to the proposed option:

```ts
test('example', async ({ page }) => {
  // ... test goes here ... 

  if (process.env.LEAVE_BROWSER_OPEN)
    await page.pause();
});
```

You can run it with `LEAVE_BROWSER_OPEN=1 npx playwright test`. Let me know whether this helps.

**Comment by f-w** [comment]
@dgozman, `--leave-browser-open` option gives test runners a more consistant user experience than your proposal because test scripts can be authored by different people. It is not trivial to enforce all scripts contain the same code block you gave. The consistant user experience is a more important concern especially when test runners are less technical savvy to have the capacity to inspect source to figure out why some scripts work as expected and others don't.

**Comment by dgozman** [comment]
@f-w You can structure the code to avoid putting this snippet everywhere. For example, by introducing an [automatic fixture](https://playwright.dev/docs/test-fixtures#automatic-fixtures), or [overriding the page fixture](https://playwright.dev/docs/test-fixtures#overriding-fixtures).

I am recommending a DIY solution to unblock you, because I don't expect this feature to be implemented soon.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.

**Comment by arekon7** [comment]
@f-w Hi, Have you solved this somehow? I really need this feature as well. 
Since in Selenium I used to use opened browser to do some other manual testing actions after the test, and now its closing and there is no way to keep it open, unless you hang the script with _pause_ or similar, but in that case the test outcome and status is not visible, which I need as well (test being stuck in running mode) .

**Comment by f-w** [comment]
@arekon7 , I solved it using automatic fixture below.

```ts
import { test as base } from '@playwright/test';

export const test = base.extend({
  leaveBrowserOpen: [
    async ({ browser, page }, use, testOptions) => {
      await use();
      const pagePauses = [];
      if (!testOptions.project.use.headless) {
        for (let c of browser.contexts()) {
          for (let p of c.pages()) {
            pagePauses.push(p.pause());
          }
        }
      }
      await Promise.all(pagePauses);
    },
    { scope: 'test', auto: true },
  ],
});
```

`testOptions.project.use.headless`, which is defined in _playwright.config.ts_, can be based on an env var to distinguish manual scenario.

**Comment by arekon7** [comment]
@f-w Thanks for sharing the code snippet! That really solves this in a way! But in my case I would like to see the test output (failed/passed) with all failed _expect_ assertions (if any). And if you pause the test at the end, the output and result state is not available. Have you considered this or that or it was not your case? Thanks!

**Comment by f-w** [comment]
@arekon7, output is available on both console and _playwright-report_ folder. The folder also contains video recording of the test.
